internal/handlers/common/msg_findandmodify.go

```diff
@@ -0,0 +1,152 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"fmt"
+
+	"github.com/FerretDB/FerretDB/internal/types"
+)
+
+// FindAndModifyParams represent all findAndModify requests' fields.
+// It's filled by calling prepareFindAndModifyParams.
+type FindAndModifyParams struct {
+	DB, Collection, Comment               string
+	Query, Sort, Update                   *types.Document
+	Remove, Upsert                        bool
+	ReturnNewDocument, HasUpdateOperators bool
+	MaxTimeMS                             int32
+}
+
+// PrepareFindAndModifyParams prepares findAndModify request fields.
+func PrepareFindAndModifyParams(document *types.Document) (*FindAndModifyParams, error) {
+	var err error
+
+	command := document.Command()
+
+	var db, collection string
+
+	if db, err = GetRequiredParam[string](document, "$db"); err != nil {
+		return nil, err
+	}
+
+	if collection, err = GetRequiredParam[string](document, command); err != nil {
+		return nil, err
+	}
+
+	if collection == "" {
+		return nil, NewErrorMsg(
+			ErrInvalidNamespace,
+			fmt.Sprintf("Invalid namespace specified '%s.'", db),
+		)
+	}
+
+	var remove bool
+
+	if remove, err = GetBoolOptionalParam(document, "remove"); err != nil {
+		return nil, err
+	}
+
+	var returnNewDocument bool
+
+	if returnNewDocument, err = GetBoolOptionalParam(document, "new"); err != nil {
+		return nil, err
+	}
+
+	var upsert bool
+
+	if upsert, err = GetBoolOptionalParam(document, "upsert"); err != nil {
+		return nil, err
+	}
```

Comment:
Just to make spacing a bit more consistent. 😃 
```suggestion
	var db string
	if db, err = GetRequiredParam[string](document, "$db"); err != nil {
		return nil, err
	}

	var collection string
	if collection, err = GetRequiredParam[string](document, command); err != nil {
		return nil, err
	}

	if collection == "" {
		return nil, NewErrorMsg(
			ErrInvalidNamespace,
			fmt.Sprintf("Invalid namespace specified '%s.'", db),
		)
	}

	var remove bool
	if remove, err = GetBoolOptionalParam(document, "remove"); err != nil {
		return nil, err
	}

	var returnNewDocument bool
	if returnNewDocument, err = GetBoolOptionalParam(document, "new"); err != nil {
		return nil, err
	}

	var upsert bool
	if upsert, err = GetBoolOptionalParam(document, "upsert"); err != nil {
		return nil, err
	}
```